### PR TITLE
Update doc to replace aks-engine with capz

### LINKED ISSUE #1927: update docs to correctly describe how to run e2e tests
#### What happened:

https://kubernetes-sigs.github.io/cloud-provider-azure/development/e2e/e2e-tests/ talks about using the now-deprecated AKS Engine for e2e tests.

#### What you expected to happen:

Because https://github.com/kubernetes-sigs/cloud-provider-azure/issues/919 is resolved, I thought that we are deprecating e2e tests that use AKS Engine.

#### How to reproduce it (as minimally and precisely as possible):

See https://github.com/kubernetes-sigs/cloud-provider-azure/blob/master/site/content/en/development/e2e/e2e-tests.md#how-to-run-kubernetes-e2e-tests-locally for the outdated information.

#### Anything else we need to know?:

Please update the docs so we don't have people thinking that AKS Engine should be or is used for e2e tests. Thanks.

### PR DESCRIPTION
**Pull request by lzhecheng**
Signed-off-by: Zhecheng Li <zhechengli@microsoft.com>

#### What type of PR is this?
/kind documentation

#### What this PR does / why we need it:
Update doc to replace aks-engine with capz
#### Which issue(s) this PR fixes:

Fixes #1927

#### Does this PR introduce a user-facing change?

```release-note
Update doc to replace aks-engine with capz
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: hack/deploy-cluster-capz.sh [hunk 1]
```diff
@@ -40,6 +40,7 @@ export AZURE_LOCATION="${AZURE_LOCATION:-westus2}"
 export AZURE_CLOUD_CONTROLLER_MANAGER_IMG="${AZURE_CLOUD_CONTROLLER_MANAGER_IMG:-mcr.microsoft.com/oss/kubernetes/azure-cloud-controller-manager:v1.23.1}"
 export AZURE_CLOUD_NODE_MANAGER_IMG="${AZURE_CLOUD_NODE_MANAGER_IMG:-mcr.microsoft.com/oss/kubernetes/azure-cloud-node-manager:v1.23.1}"
 export KUBERNETES_VERSION="${KUBERNETES_VERSION:-v1.23.0}"
+export IN_TREE="${IN_TREE:-false}"
```
Comment (nilo19) [inline hack/deploy-cluster-capz.sh:43]: Can we make it `USE_IN_TREE_CLOUD_PROVIDER`?

File: site/content/en/development/e2e/e2e-tests-azure.md [hunk 2]
```diff
@@ -37,14 +37,16 @@ export AZURE_LOADBALANCER_SKU=<loadbalancer-sku> # the sku of load balancer (opt
...
-    For more usage of ginkgo, please follow [ginkgo](https://github.com/onsi/ginkgo/blob/master/README.md)
+  The following command ensures [gingko](https://github.com/onsi/ginkgo) v2 is installed and then runs default tests.
 
-### Without ginkgo
-- Run ```go test ./tests/e2e/ -timeout 0```
+  ``` make test-ccm-e2e ```
+
+- Run specific tests
+
+  ``` go test -v ./tests/e2e/ -timeout 0 -ginkgo.focus <focus-keyword> --ginkgo.skip <skip-keyword> ```
```
Comment (nilo19) [inline site/content/en/development/e2e/e2e-tests-azure.md:50]: Have you checked this with ginkgo v2? It looks there is no ginkgo.focus and ginkgo.skip any more.

File: hack/deploy-cluster-capz.sh (hunk 1 above)
Comment (lzhecheng) [reply hack/deploy-cluster-capz.sh]: Updated.

File: site/content/en/development/e2e/e2e-tests-azure.md (hunk 2 above)
Comment (lzhecheng) [reply site/content/en/development/e2e/e2e-tests-azure.md:50]: Yes, with TOT code, go test with such option works.

**Pull request merged**